src/Page/Form.purs

```diff
@@ -206,26 +154,39 @@ component =
         , HE.onChecked (HE.input $ UpdateKey field.key <<< Boolean)
         ]
 
-  getValue
-    :: ∀ r
-     . Record (SharedRows ExprType r)
-    -> Maybe ExprType
-  getValue { value: UserInput x } = Just x
-  getValue { default } = default
-
-  setPage :: Key -> ExprType -> Page Expr -> Page Expr
-  setPage key val page =
-    page { contents = map (setSection key val) page.contents}
-
-  setSection :: Key -> ExprType -> Section Expr -> Section Expr
-  setSection key val section =
-    section { contents = map (setField key val) section.contents}
-
-  setField :: Key -> ExprType -> Field Expr -> Field Expr
-  setField key val field
-    | key == field.key = case field.input of
-      Text input ->
-        field { input = Text input { value = UserInput (Val val) } }
-      Toggle input ->
-        field { input = Toggle input { value = UserInput (Val val) } }
-    | otherwise = field
+eval :: forall m. Query ~> H.ParentDSL State Query (ChildQuery m) ChildSlot Message m
+eval = case _ of
+  Initialize a -> a <$ do
+    H.modify_ _ { values = Lynx.Data.Form.keys testPage }
+    void $ bitraverse_
+      (\e -> H.modify _ { form = Failure e })
+      (\f -> eval $ EvalForm f a)
+      (Right testPage)
+
+  EvalForm expr a -> a <$ do
+    { values } <- H.get
+    let evaled' = Lynx.Data.Form.eval (\key -> Data.Map.lookup key values) expr
+    evaled <- for evaled' \page -> do
+      for_ page.contents \section -> do
+        for_ section.contents \field -> do
+          case field.input of
+            Dropdown dropdown ->
+              for_ (toArray dropdown.options) \options ->
+                H.query' cp1 field.key (Dropdown.SetItems options unit)
```

Comment:
Ooph, sorry to hit you with another one. We're not currently handling the case where an invalid option is selected. So if you load up the current form, select "Strawberry" from the food dropdown, then toggle "Active" to off, "Strawberry" remains selected despite being no longer valid.

I would say the behavior here hasn't really been specified, so I'm okay if you want to merge this in and then tackle this behavior in another PR.

But here the value on the form state should probably become `Invalid (String "Strawberry")`, and the value in the dropdown should be reset to `Nothing`. Glad to discuss more next week.

Reply:
Since we don't do any validation for any of the fields, it seems like we should try to tackle that separately. I think we should punt on this.